spec/Candid.md

```diff
@@ -1253,9 +1253,43 @@ M(ref(r) : principal) = i8(0)
 M(id(v*) : principal) = i8(1) M(v* : vec nat8)
 ```
 
-Note:
+##### Wire cost model
+
+Some values on the wire take fewer space than the values on the host language, e.g., `null`, `reserved` and `record { record {} }` take no space on the wire. To prevent sending messages of exponential size, we introduce a cost model `C` for the values on the wire.
+
+```
+C : <val> -> <primtype> -> nat
+C(n : nat)      = 1
```

Comment:
The cost for nat shouldn't be constant Can we not use the natural bit or byte size for each value?

Reply:
Yeah, we should. I was undecided whether the cost should be `log2 n` or the actual bytes in LEB128. One can keep padding `\x80` in the LEB128 encoding with a small `n`. Also, should we charge the same for length encoding in the constructor types? It was currently 1.